crates/dekaf/src/session.rs

```diff
@@ -847,6 +847,14 @@ impl Session {
             .connect_to_group_coordinator(req.group_id.as_str())
             .await?;
 
+        let flow_client = self
+            .auth
+            .as_mut()
+            .ok_or(anyhow::anyhow!("Session not authenticated"))?
+            .authenticated_client()
+            .await?
+            .clone();
```

Comment:
AFAICT, it's needed either here or inside `authenticated_client()`. `authenticated_client()`'s signature desugars to 

``` rust
pub async fn authenticated_client<'a>(&'a mut self) -> anyhow::Result<&'a flow_client::Client>
```

AFAICT, because the returned reference `&'a flow_client::Client`, has the same lifetime as the mutable reference to `self`, it too represents a mutable borrow. This causes problems later on in `offset_commit` where we call `self.decrypt_topic_name` which takes a `&self`, which is disallowed because the `flow_client` holds a mut reference with lifetime >= `&mut self`, and we get the error

```
cannot borrow `*self` as immutable because it is also borrowed as mutable
immutable borrow occurs here // call to self.decrypt_topic_name()
session.rs(850, 27): mutable borrow occurs here // call to self.auth.authenticated_client()
session.rs(873, 57): mutable borrow later used here // call to Collection::new(&flow_client)
```

I could change `authenticated_client()` to return `anyhow::Result<flow_client::Client>`, but that would just mean `.clone()`ing the client in there instead. Maybe that's better ergonomically?

Reply:
Nope, this is good. I just didn't recall that it already returned a reference. IMO it's better to let the caller decide whether to take a clone or use a reference, rather than taking that choice from them.